### PR TITLE
feat(pages): add comparison between callcode and delegatecall

### PR DESCRIPTION
**Pull request by wiasliaw**

### CODE REVIEW

**Comment by engn33r** [comment]
thanks for the PR :+1: I will look it over closely in the coming days before merging

**Comment by engn33r** [comment]
I would like to make some changes to this PR unless you have responses to these points:
- The comparison you make is very confusing to read. Did you use ChatGPT to generate it? I was able to generate a similar sounding result that also did not sound very clear. I would suggest using `msg.value` instead of "value" and `msg.sender` instead of "caller". You even used this in the diagram.
- I don't see much benefit to introducing the terms "sub-context", "code region", and "storage region". These are not needed to explain the differences between the opcodes. Let's keep things simple and remove unnecessary information. If someone wants a deep dive on how delegatecall works, they can find other resources.
- The last sentence containing "`callcode` can be used in Proxy Pattern" -> this is incorrect for solidity versions 0.5.0 and after. I suggest deleting this sentence and the one after it because we should not be encouraging old solidity versions to be used. The earlier comparison of `callcode` and `delegatecall` already indicates they are very similar.

In summary, I like the diagram and will happily merge it. But most of the other new text sounds overly complex and repeats text already in the page. If there is a sentence or two that has unique information, it would be easier to add it into the existing text instead of making a new section.

**Comment by wiasliaw** [comment]
- I only used chatGPT in grammar correction. I might check it by my self.
- I'll make execution context to further reading.
- The reason of `callcode` using in proxy is to emphasize the both `callcode` and `delegatecall` have the same behavior on storage. BTW, even though `address.callcode` is deprecated, it still available in assembly like `assembly { callcode() }`.

I'll rewrite the whole part asap.

**Comment by devtooligan** [comment]
@engn33r asked me to give this a second look

Personally I like the writeup. While not used much anymore, it's important to understand callcode.  In fact it came up in a recent audit I was on.

I dont mind the use of caller and value but I could see how that might add some confusion so might as well change it.

 I agree w the suggestion of not mentioning "sub-context" and the other terms.  That would be good to talk about in a blogpost - which I would encourage you to consider doing to increase awareness of callcode!

I also agree, lets not mention proxy at the bottom, we can remove that.

Thanks so much for submitting this pr!! 🫡

**Comment by engn33r** [comment]
Thank you for the PR @wiasliaw! Old opcodes are not talked about much so it helps to clarify this topic.
